src/main/java/io/jenkins/plugins/conventionalcommits/utils/GetProjectType.java

```diff
@@ -0,0 +1,13 @@
+package io.jenkins.plugins.conventionalcommits.utils;
+
+import java.io.File;
+import hudson.FilePath;
+
+public class GetProjectType {
+
+    public boolean checkMaven(File directory) {
```

Comment:
This class should be just like your `CurrentVerion` class: this should have 1 public method that returns the type of project with private methods for each of the different project types.